twisted/test/test_tcp.py

```diff
@@ -1138,11 +1140,18 @@ def getHandleErrorCode(self):
         Return the errno expected to result from writing to a closed
         platform socket handle.
         """
-        # These platforms have been seen to give EBADF:
-        #
-        #  Linux 2.4.26, Linux 2.6.15, OS X 10.4, FreeBSD 5.4
-        #  Windows 2000 SP 4, Windows XP SP 2
-        return errno.EBADF
+        if platform.isWindows():
```

Comment:
I think this could be simplified to:

```python
# Windows and Python 3: WSAENOTSOCK
# Windows and Python 2: EBADF
# Linux, FreeBSD, Mac OS X: return EBADF
if platform.isWindows() and _PY3:
    return errno.WSAENOTSOCK
return errno.EBADF
```

Note that all the listed Linux, OS X, and FreeBSD versions are ancient and now irrelevant.